engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py

```diff
@@ -22,6 +22,11 @@ def resolve_alert_group_by_source_if_needed(alert_group_pk):
             alert_group.active_resolve_calculation_id
         )
     else:
+        if alert_group.resolved_by == alert_group.NOT_YET_STOP_AUTORESOLVE:
+            return "alert_group is too big to auto-resolve"
+        if alert_group.alerts.count() > AlertGroupForAlertManager.MAX_ALERTS_IN_GROUP_FOR_AUTO_RESOLVE:
```

Comment:
if more than N alerts in alert group, set the flag and make the last attempt to autoresolve

Reply:
I like this design!